### PR TITLE
chore: Bug fixes in the kafka instance table kabab menu

### PR DESCRIPTION
**Pull request by hemahg**
**What this PR does / why we need it**:

- [x] Fixes the Change Owner kabab option doesn’t show on the data plane UI.
- [x] Fixes The connection kabab option is disabled for an instance being created.
- [x] Add function to check if the user has permission to  Delete an Instance
- [x] Add function to check if the user has permission to change owner of an instance
 
**Which issue(s) this PR fixes** 

> Replace this comment with references to related issues.
> e.g. `fixes #<issue number>`

**Verification steps** 

> Attach a screenshot of the output.
> guidance on how to go the particular story.
> e.g. `point to the story with a text breadcrumb, something like "components > <component name> > <story name>”`

**Is there any work left / what's next** :

> Replace this comment with pending items that need to be covered in this PR or some pending items you are planning to do in the future PR.

**Special notes for your reviewer**:

### CODE REVIEW

File: apps/consoledot-rhosak/src/routes/control-plane/routes/KafkaInstancesRoute.tsx [hunk 1]
```diff
@@ -40,6 +43,20 @@ export const KafkaInstancesRoute: FunctionComponent<KafkaInstancesRoute> = ({
...
+  const [isOrgAdmin, setIsOrgAdmin] = useState<boolean>(false);
+
+  useEffect(() => {
+    const getUser = async () => {
+      const user = await auth.getUser();
+      setCurrentUser(user?.identity?.user?.username ?? "");
+      setIsOrgAdmin(user?.identity?.user?.is_org_admin ?? false);
+    };
+    void getUser();
+  }, [auth]);
```
Comment (riccardo-forina) [inline apps/consoledot-rhosak/src/routes/control-plane/routes/KafkaInstancesRoute.tsx:58]: I'd make this a fetcher in the api, with its own useUser hook. So it's cached and it's ready to be reused.

Just remember to add the new dependency on the api package.

File: apps/consoledot-rhosak/src/routes/data-plane/routes/DataPlaneTopicHeaderConnected.tsx [hunk 2]
```diff
@@ -56,10 +62,10 @@ export const DataPlaneTopicHeaderConnected: VoidFunctionComponent<
         toggleExpanded(true);
       }}
       canOpenConnection={
-        instance ? ReadyStatuses.includes(instance?.status) : false
+        instance ? !SuspendedStatuses.includes(instance?.status) : false
```
Comment (riccardo-forina) [inline apps/consoledot-rhosak/src/routes/data-plane/routes/DataPlaneTopicHeaderConnected.tsx:65]: All these includes are logic that is spread across the app. We should have a single function that contains this logic.

File: packages/consoledot-api/src/instancePermission.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import type { KafkaInstanceEnhanced } from "consoledot-api/src/transformers/kafkaRequestToKafkaInstanceEnhanched";
+import { SuspendedStatuses } from "ui-models/src/models/kafka";
+import type { Status } from "ui-models/src/models/kafka";
+
+export function canOpenConnection(instance: KafkaInstanceEnhanced): boolean {
+  return instance ? !SuspendedStatuses.includes(instance.status) : false;
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/instancePermission.ts:6]: Shouldn't this logic include only the ready states? I can't connect to a failed or deleted instance, right?

Comment (hemahg) [reply packages/consoledot-api/src/instancePermission.ts:6]: @riccardo-forina , I checked the old code, connection action in the kebab menu will be disable when the instance status is suspended

Comment (riccardo-forina) [reply packages/consoledot-api/src/instancePermission.ts:6]: I'm not sure we should assume the old code was right. Actually, I'd urge you to challenge it all the time :)

Let's reason about this: what good does having the URL to connect to an instance that's being deleted? Or that's failed? Isn't the only safe moment to connect to the instance when the instance is ready?

Comment (hemahg) [reply packages/consoledot-api/src/instancePermission.ts:6]: When we are deleting the instance, kebab actions will not be visible

Comment (riccardo-forina) [reply packages/consoledot-api/src/instancePermission.ts:6]: That doesn't mean the logic can be flawed.

Comment (hemahg) [reply packages/consoledot-api/src/instancePermission.ts]: @riccardo-forina I reverted the logic to include only the ready state, But the only concern I have is the connection option will be disabled, do we need to show some message about why the option is disabled, and also the user can still access the connection tab by click on the details option will that be a good option?

Comment (riccardo-forina) [reply packages/consoledot-api/src/instancePermission.ts]: If the option in the kebab menu is disabled, the tab should be disabled as well. A tooltip to explain why the tab/action is disabled is a good idea but maybe let's do it in a future PR

File: packages/consoledot-api/src/queries/UseUser.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    queryFn: async () => {
+      const user = await auth.getUser();
+      return {
+        username: user?.identity?.user?.username ?? "",
+        isOrgAdmin: user?.identity?.user?.is_org_admin ?? false,
+      };
+    },
+  });
+
+  return data ?? { username: "", isOrgAdmin: false };
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/queries/UseUser.ts:18]: I'm not a fan of this. It's better to return the returned value of `useQuery` directly, so we have access to all the other properties that the hook gives us (all the `isXXX` booleans for example).

As I commented elsewhere to Suyash, the fact that data might be unavailable is a problem we have to deal with, we can't just default values and call it a day.

File: packages/consoledot-api/src/queryKeys.ts [hunk 5]
```diff
@@ -18,6 +18,7 @@ export const masQueries = {
     [{ ...masQueries._root(), entity: "organization" }] as const,
   userAccounts: (params: Omit<FetchUserAccountsParams, "getUserAccounts">) =>
     [{ ...masQueries._root(), entity: "userAccounts" }, params] as const,
+  userName: () => [{ ...masQueries._root(), entity: "userName" }] as const,
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/queryKeys.ts:21]: I'd just name this `user`, as it's an object with extra info about him.

File: apps/consoledot-rhosak/src/utils.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+export const canChangeOwner = (
+  owner: string,
+  status: Status,
+  username: string,
+  isOrgAdmin: boolean
+) => {
+  return (
+    owner === username || (isOrgAdmin && !SuspendedStatuses.includes(status))
+  );
+};
```
Comment (riccardo-forina) [inline apps/consoledot-rhosak/src/utils.ts:26]: As we now have a "user" type returned by `useUser`, maybe we can think of returning these functions from that hook, so we don't have to pass the username around.

File: packages/consoledot-api/src/queries/UseUser.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+      (data.isOrgAdmin && !SuspendedStatuses.includes(status))
+    );
+  };
+
+  return {
+    ...data,
+    canOpenConnection,
+    isUserOwnerOrAdmin,
+    canChangeOwner,
+  };
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/queries/UseUser.ts:67]: Same comment I have made elsewhere. We need to handle the loading off the data, we can't just default all. Add the usare is a critical bit of data, this should be a gate. And all the object construction should happen inside the useQuery, not outside of it.

File: packages/consoledot-api/src/queries/index.ts [hunk 8]
```diff
@@ -9,3 +9,4 @@ export * from "./useUserAccounts";
 export * from "./useConsumerGroup";
 export * from "./useAcls";
 export * from "./useServiceAccounts";
+export * from "./UseUser";
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/queries/index.ts:12]:
```suggestion
export * from "./useUser";
```

The filename should have the same case as the exported function.

**Review by riccardo-forina — CHANGES_REQUESTED** [review]

**Review by riccardo-forina — APPROVED** [review]

**Pull request merged**